tests/posix_close/main.c

```diff
@@ -0,0 +1,77 @@
+#include <errno.h>
+#include <fcntl.h>
+#include <pthread.h>
+#include <signal.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+
+#include "parsenum.h"
+#include "posix_close.h"
+#include "warnp.h"
+
+#include "open_close.h"
+#include "send_alarms.h"
+
+int
+main(int argc, char * argv[])
+{
+	const char * filename;
+	int numbytes;
+	volatile int stop_open_close = 0;
+	volatile int stop_alarms = 0;
+	int rc;
+	pthread_t open_close_thread;
+	pthread_t send_alarms_thread;
+
+	WARNP_INIT;
+
+	/* Process arguments. */
+	if (argc != 3) {
+		fprintf(stderr, "usage: test_posix_close filename numbytes\n");
+		goto err0;
+	}
+	filename = argv[1];
+	if (PARSENUM(&numbytes, argv[2], 0, 1)) {
+		warn0("%s should be 0 or 1", argv[2]);
+		goto err0;
+	}
+
+	/* Open, read 0 or 1 bytes, then close the file. */
+	if (start_repeated_open_close(filename, numbytes, &stop_open_close,
+	    &open_close_thread))
+		goto err0;
+
+	warn0("sleep for first thread to start");
```

Comment:
The question here is "how to wait for the thread to start?".  As you might recall, I wrote a `pthread_create_blocking()` that we didn't end up using in `spipe`.  That seems a bit like overkill, though, especially since this is just a test and there's no harm (?) in slapping a `sleep(1)` in here.  (Although 1 full second is awfully long.)